Switch over CoreCLR pr.yml to use live-built libraries

trylek opened the pull request.

trylek committed:
Switch over CoreCLR pr.yml to use live-built libraries

trylek committed:
Delete CoreFX runs against CoreCLR

safern commented:
FYI: @BruceForstall this will remove the CoreFX stress legs running against a checked CoreCLR and we will need to bring those back later once all pipeline changes are settled.

safern commented on eng/pipelines/coreclr/pr.yml:
```diff
@@ -42,6 +42,20 @@ jobs:
+- template: /eng/pipelines/common/platform-matrix.yml
+  parameters:
+    jobTemplate: /eng/pipelines/libraries/build-job.yml
+    buildConfig: Debug
```
Does it really matter having Debug vs Release builds of the libraries product? The API surface area doesn't change at all. 

I don't fully know what it is used on CoreCLR for, but it might not be needed to have Debug builds as well.

jashook replied:
I think we should push to always use a release build of libraries in coreclr, even for a debug build of coreclr. If plumbing this through is more work at the moment then we can work on this after the change goes in.

safern replied:
I don't think it shouldn't be much work. It is just a matter of including this platforms on the Release template expansion and passing down the correct parameter for liveCoreClrConfiguration set to Release.

safern replied:
Actually there's already a Release builds section for libraries which is building these two platforms, so we should just update all coreclr build that depend on libraries to use Release bits.

trylek replied:
Thanks for pointing that out, fixed.

BruceForstall commented:
> FYI: @BruceForstall this will remove the CoreFX stress legs running against a checked CoreCLR and we will need to bring those back later once all pipeline changes are settled.

Thanks for the heads-up. I knew this was coming.

jashook commented:
ci.yml will need the same changes correct?

jashook commented:
I also assume all the other stress pipelines will need to be modified to be live live as well.

jashook commented:
If that is the case ci.yml is the highest priority, the others are not blocking.

safern reviewed (approved):
One comment, otherwise, LGTM.

Also, yes we should update ci.yml as part of this change or a follow-up change.

trylek committed:
PR feedback - only use release CoreFX builds for CoreCLR testing

trylek committed:
Remove OSSubgroup from CoreFX build artifact name

safern commented on eng/pipelines/coreclr/templates/xplat-pipeline-job.yml:
```diff
@@ -109,7 +109,7 @@ jobs:
 
     - ${{ if ne(parameters.liveLibrariesBuildConfig, '') }}:
-      - librariesArtifactName: ${{ format('libraries_bin_{0}_{1}{2}_{3}_{4}', 'netcoreapp', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
+      - librariesArtifactName: ${{ format('libraries_bin_netcoreapp_{0}_{1}_{2}', parameters.osGroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
```
Why are we removing the osSubgroup? wouldn't this break the linux_musl libraries builds?

safern replied:
Oh sorry. I see this is just in coreclr's consumption, but this might break if you guys start depending on an artifact that contains ossubgroup like musl.

trylek replied:
Hmm, I was under the impression you yourself told me the other day that netcoreapp artifacts don't include OS subgroup information.

trylek committed:
Clean up propagation of library artifacts per Santi's feedback

safern commented on eng/pipelines/common/xplat-setup.yml:
```diff
@@ -49,6 +49,9 @@ jobs:
     osSubgroup: ${{ parameters.osSubgroup }}
     archType:  ${{ parameters.archType }}
 
+    librariesBuildArtifactName: ${{ format('libraries_bin_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
```
Oopps. This is not a variable, this is defined as a parameter. It seems like you want it under the variables section.

safern replied:
Thanks for doing the clean-up btw.

trylek replied:
Right you are, thanks for spotting, hopefully fixed now.

trylek replied:
Hmm, it looks like you actually sent me down a pretty nasty rathole. I'm right now trying to figure out how to consolidate the movement of buildConfig to xplat-setup with the <code>liveLibrariesBuildConfig</code> parameter but the hope to get this merged in today just keeps fading away.

trylek replied:
Long story short, having an independent configuration pair for CoreCLR and CoreFX is not an ideal start for a common xplat-setup script used by both.

trylek replied:
For now I'm inclined to roll back all the "library artifact cleanup changes" as I see no other way how to fix the pr.yml script.

safern replied:
sounds good to me and sorry about that. We can do that clean-up later. 

Sorry for sending you that path, didn't think it was going to be that hard.

trylek replied:
No worries, it also came as a surprise to me. But yes, I think it's better to postpone the cleanup after things have settled down, we're still pretty much shooting at a moving target right now.

safern replied:
Agreed.

trylek committed:
Fix typo in placement of the artifact names (thanks Santi for spotting!)

trylek committed:
Make xplat-setup aware of buildConfig

This is needed so that the library name artifacts are properly
synthesized. This extra delta is somewhat noisy but hopefully trivial.

Thanks

Tomas

trylek committed:
Rollback cleanup of library artifact naming as it doesn't work
